src/library/scala/collection/package.scala

```diff
@@ -77,4 +77,12 @@ package object collection {
       if(t.isEmpty) None
       else Some(t.init -> t.last)
   }
+
+  /** Common message for IndexOutOfBounds exceptions with a known max index */
+  private[collection] def indexOutOfBoundsMessage(index: Int, max: Int): String = 
```

Comment:
maybe mark the helpers `@noinline`, as this is the goal here?

Reply:
Thanks. Done in [face23b](https://github.com/scala/scala/pull/10928/commits/face23bb69beb5b471c29e604cf8a3224e35d538).